compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/UIKitComposeSceneLayer.uikit.kt

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.scene
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionContext
+import androidx.compose.ui.graphics.Canvas
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Paint
+import androidx.compose.ui.input.key.KeyEvent
+import androidx.compose.ui.input.pointer.PointerButton
+import androidx.compose.ui.input.pointer.PointerEventType
+import androidx.compose.ui.platform.PlatformContext
+import androidx.compose.ui.platform.PlatformWindowContext
+import androidx.compose.ui.uikit.ComposeUIViewControllerConfiguration
+import androidx.compose.ui.uikit.density
+import androidx.compose.ui.uikit.layoutConstraintsToMatch
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.unit.asDpOffset
+import androidx.compose.ui.unit.asDpSize
+import androidx.compose.ui.unit.round
+import androidx.compose.ui.unit.toOffset
+import androidx.compose.ui.window.FocusStack
+import androidx.compose.ui.window.GestureEvent
+import androidx.compose.ui.window.MetalView
+import kotlin.coroutines.CoroutineContext
+import kotlinx.cinterop.CValue
+import kotlinx.cinterop.useContents
+import platform.CoreGraphics.CGPoint
+import platform.UIKit.NSLayoutConstraint
+import platform.UIKit.UIView
+
+// TODO: perhaps make LayerComposeSceneMediator a ComposeSceneLayer?
+internal class UIKitComposeSceneLayer(
+    private val onClosed: (UIKitComposeSceneLayer) -> Unit,
+    private val createComposeSceneContext: (PlatformContext) -> ComposeSceneContext,
+    private val providingCompositionLocals: @Composable (@Composable () -> Unit) -> Unit,
```

Comment:
Having such arg is highly questionable, but it requires more time to figure out details/suggest replacement

Reply:
It's just a shadow of former design, where we somehow need someone else to provide our composition locals.